test/model/content_test.dart

```diff
@@ -1,16 +1,429 @@
+    'code block, with an unknown span type',
+    null, // this test is for future Pygments versions adding new token types
+    '<div class="codehilite" data-code-language="Dart"><pre>'
+        '<span></span><code><span class="unknown">class</span>'
+        '\n</code></pre></div>', [
+      blockUnimplemented('<div class="codehilite" data-code-language="Dart"><pre>'
+        '<span></span><code><span class="unknown">class</span>'
+        '\n</code></pre></div>'),
+    ]);
+
+  static final mathInline = ContentExample.inline(
+    'inline math',
+    r"$$ \lambda $$",
+    expectedText: r'\lambda',
+    '<p><span class="katex">'
+      '<span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>λ</mi></mrow>'
+        '<annotation encoding="application/x-tex"> \\lambda </annotation></semantics></math></span>'
+      '<span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6944em;"></span><span class="mord mathnormal">λ</span></span></span></span></p>',
+    const MathInlineNode(texSource: r'\lambda'));
+
+  static const mathBlock = ContentExample(
+    'math block',
+    "```math\n\\lambda\n```",
+    expectedText: r'\lambda',
+    '<p><span class="katex-display"><span class="katex">'
+      '<span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mi>λ</mi></mrow>'
+        '<annotation encoding="application/x-tex">\\lambda</annotation></semantics></math></span>'
+      '<span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6944em;"></span><span class="mord mathnormal">λ</span></span></span></span></span></p>',
+    [MathBlockNode(texSource: r'\lambda')]);
+
+  static const mathBlockInQuote = ContentExample(
+    'math block in quote',
+    // There's sometimes a quirky extra `<br>\n` at the end of the `<p>` that
+    // encloses the math block.  In particular this happens when the math block
+    // is the last thing in the quote; though not in a doubly-nested quote;
+    // and there might be further wrinkles yet to be found.  Some experiments:
+    //   https://chat.zulip.org/#narrow/stream/7-test-here/topic/content/near/1715732
+    "````quote\n```math\n\\lambda\n```\n````",
+    '<blockquote>\n<p>'
+      '<span class="katex-display"><span class="katex">'
+        '<span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mi>λ</mi></mrow>'
+          '<annotation encoding="application/x-tex">\\lambda</annotation></semantics></math></span>'
+        '<span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6944em;"></span><span class="mord mathnormal">λ</span></span></span></span></span>'
+      '<br>\n</p>\n</blockquote>',
+    [QuotationNode([MathBlockNode(texSource: r'\lambda')])]);
+
+  static const singleImage = ContentExample(
```

Comment:
There's going to be a lot of statics on this class (there are already kind of a lot), so it'll need some care to help keep it possible to find the example one is looking for. One thing that I think will help is to make related examples all sort together alphabetically, by putting at the start of the name the word that describes the subject area of the example. So in particular I'd like to have all the examples in this PR take names starting with "image", even at the cost of making the names read less like natural English phrases.

For this one:

```suggestion
  static const image = ContentExample(
```

or could also be `imagePlain` or `imageSingle`.